src/core/scheduler/scheduler.cc

```diff
@@ -228,6 +229,60 @@ void Graph::Debug() {
   printf("%s", ss.str().c_str());
 }
 
+void Graph::PrintTimeProfiling() {
+  std::stringstream ss;
+
+  // verbosity level: 1 -> forward and backward propagation time
+  if (device_->verbosity() == 1) {
+    bool forward = true;
+    float forward_time = 0;
+    float backward_time = 0;
+    float time_elapsed;
+
+    for (size_t i = 0; i < nodes_.size(); ++i)
+      if (nodes_[i]->time_elapsed() > 0) {
+        if (forward == true)
+          // check the op of cross entropy backward, after that are backward ops
+          if (nodes_[i]->op_name().find("Backward") != std::string::npos)
```

Comment:
Do all backward propagation meet this condition?

Reply:
I have added the note, it is more accurate when either "SoftmaxCrossEntropy" or "Softmax" is used, because both backward operation has "Backward" in the name as the dividing line between the forward and backward.